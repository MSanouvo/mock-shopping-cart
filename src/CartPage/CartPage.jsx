import NavBar from "../components/Navbar/NavBar";
import CartItem from "../components/cartItem/cartItem";


function CartPage () {
	return (
		<div id="container">
			<NavBar />
            <h2>Shopping Cart</h2>
			<div className="store-content">
				<CartItem name="item1" count={1} handleClick={()=>{}} />
				<CartItem name="item2" count={1} handleClick={()=>{}} />
				<CartItem name="item3" count={1} handleCLick={()=>{}} />
			</div>
		</div>
	)
}

export default CartPage