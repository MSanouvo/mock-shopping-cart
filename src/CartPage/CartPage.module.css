.container{
    /* border: 1px solid yellow; */
    min-width: 100vw;
}

.heading{
    font-weight: bold;
    font-size: 2.5rem;
    padding: 20px;
    color: black;
}

.cart{
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: start;
    gap: 20px;
    padding: 10px 20px;
    color: black;
    /* outline: 1px solid black; */
}

.cartItems{
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    gap: 30px;
    box-shadow: 0px 0px 5px black;
}

.button{
    display: flex;
    /* border: 1px solid green; */
    padding: 30px;
    width: 95%;
    justify-content: end;
}

.message{
    width: 100%;
    text-align: center;
}