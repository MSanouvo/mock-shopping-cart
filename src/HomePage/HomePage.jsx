import NavBar from "../components/Navbar/NavBar"
import Banner from "../assets/gunpla_banner.jpg"
import styles from "./HomePage.module.css"

function HomePage () {
    return (
        <div>
            <NavBar />
            {/* Could turn this into a component if we want to reuse it on the store page */}
            <img className={styles.banner} src={Banner} alt="store banner" />
            <main>Lorem Ipsum, ...</main>
        </div>
    )
}

export default HomePage