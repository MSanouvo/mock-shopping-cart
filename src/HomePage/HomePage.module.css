.banner {
  height: 50vh;
}

.container {
  /* border: 1px solid yellow; */
  min-width: 100vw;
}

.content {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 10px;
  padding: 30px 50px;
}

.message {
  font-size: 2rem;
  color: black;
  text-align: center;
}

.text {
  font-size: 1.2rem;
  width: 70vw;
}
