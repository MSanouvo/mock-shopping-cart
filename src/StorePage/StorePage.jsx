import NavBar from "../components/Navbar/NavBar";
import ItemCard from "../components/itemCard/itemCard";


function StorePage () {
	return (
		<div id="container">
			<NavBar />
            <h2>Store</h2>
			<div className="store-content">
				<ItemCard name="item1" count={1} handleClick={()=>{}} />
				<ItemCard name="item2" count={1} handleClick={()=>{}} />
				<ItemCard name="item3" count={1} handleCLick={()=>{}} />
			</div>
		</div>
	)
}

export default StorePage