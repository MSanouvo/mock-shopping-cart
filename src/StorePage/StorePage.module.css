.container{
    /* border: 1px solid yellow; */
    display: flex;
    flex-direction: column;
    height: 100vh;
    min-width: 100vw;
    overflow-y: scroll;
}

.heading{
    font-weight: bold;
    font-size: 2.5rem;
    padding: 20px;
    color: black;
}

.store{
    display: flex;
    flex-direction: column;
    padding: 10px 20px;
}

.items{
    /* outline: 1px solid purple; */
    display: grid;
    grid-template-rows: 100px, 1fr;
    grid-template-columns: repeat(3, 30%);
    justify-content: center;
    height: 500px;
    gap: 30px;
    grid-row: 2;
    padding: 10px;
}

.message{
    width: 100%;
    text-align: center;
}