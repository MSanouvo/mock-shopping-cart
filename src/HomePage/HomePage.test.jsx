import { describe, it, expect } from "vitest";
import userEvent from "@testing-library/user-event";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { Route, Routes } from "react-router-dom";
import HomePage from "./HomePage";
import StorePage from "../StorePage/StorePage";
import CartPage from "../CartPage/CartPage";

describe("Home page", () => {
    it("renders the Home Page", () => {
      const { container } = render(
        <MemoryRouter>
          <HomePage />
        </MemoryRouter>,
      );
      expect(container).toMatchSnapshot();
    });

  it("renders the nav bar component", () => {
    render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>,
    );
    expect(screen.getByRole("heading", { name: "Home" })).toBeInTheDocument();
    expect(screen.getByRole("heading", { name: "Shop" })).toBeInTheDocument();
    expect(screen.getByAltText("cart icon")).toBeInTheDocument();
  });

  it("renders the correct banner image", () => {
    render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>,
    );
    expect(screen.getByAltText("store banner")).toHaveAttribute("src", "/src/assets/clothes-line.jpg");
  });

  it("renders description", () => {
    render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>,
    );
    expect(screen.getByRole("main")).toHaveTextContent("Welcome to my mock digital storefront!");
  });

  it("navigates to store page", async () => {
    render(
      <MemoryRouter initialEntries={["/"]}>
        <Routes>
          <Route path="/store" element={<StorePage />} />
          <Route path="/" element={<HomePage />} />
        </Routes>
      </MemoryRouter>,
    );

    const user = userEvent.setup();
    const link = screen.getByRole("link", { name: "Shop" });

    await user.click(link);
    expect(screen.getByText("Store")).toBeInTheDocument();
  });

  it("navigates to cart page", async () => {
    render(
      <MemoryRouter initialEntries={["/"]}>
        <Routes>
          <Route path="/" element={<HomePage />} />
          <Route path="/cart" element={<CartPage cart={[]} />} />
        </Routes>
      </MemoryRouter>,
    );

    const user = userEvent.setup();
    const link = screen.getByAltText("cart icon");

    await user.click(link);
    expect(screen.getByText("Shopping Cart")).toBeInTheDocument();
  });
});
