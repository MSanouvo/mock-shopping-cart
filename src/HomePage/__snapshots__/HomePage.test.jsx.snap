// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home page > renders the Home Page 1`] = `
<div>
  <div>
    <div
      class="_navBar_1865de"
    >
      <a
        data-discover="true"
        href="/"
      >
        <h1>
          Home
        </h1>
      </a>
      <a
        data-discover="true"
        href="/store"
      >
        <h3>
          Shop
        </h3>
      </a>
      <a
        data-discover="true"
        href="/cart"
      >
        <h3>
          Cart
        </h3>
      </a>
    </div>
    <img
      alt=""
      src=""
    />
    <main>
      Lorem Ipsum, ...
    </main>
  </div>
</div>
`;
